vpos/sdk/python/bancardconnectorpython/api.py

```diff
@@ -0,0 +1,462 @@
+			:raises BancardAPIPaymentRejectecUnknownReasonException: if the payment has been rejected by Bancard due to an unhandled/unknown reason
+		"""
+
+		BancardAPI.validate_marketplace_charge_id(marketplace_charge_id)
+		BancardAPI.validate_amount(amount)
+		BancardAPI.validate_currency(currency)
+
+		amount_str = "%s.00" % currency_decimal_to_string(currency, amount)
+
+		bancard_token = "%s%s%s" % (self.private_key, marketplace_charge_id, "get_confirmation")
+		if is_python_version_greater_igual_than_3x():
+			bancard_token = bytes(bancard_token, "UTF-8")
+
+		bancard_body_request = {
+			"public_key": self.public_key,
+			"operation": {
+				"token": hashlib.md5(bancard_token).hexdigest(),
+				"shop_process_id": marketplace_charge_id
+			}
+		}
+
+		bancard_response = BancardAPI.__call_bancard_webservice(bancard_body_request, self.urls[CONFIRMATIONS_KEY])
+		if bancard_response.get("status", None) == "success":
+			confirmation = bancard_response.get("confirmation", dict())
+			response_code = confirmation.get("response_code", None)
+			if response_code == '00':
+				if Decimal(confirmation["amount"]) == Decimal(amount_str) and confirmation["currency"] == currency:
+					return True, confirmation.get("authorization_number", None), bancard_response
+				else:
+					# this marketplace charge id was found in Bancard but is has another amount/currency
+					raise BancardAPIChargeInconsistentValuesException("Duplicated charge ID %s in Bancard rejected due to inconsistency in amount/currency", bancard_response)
+			else:
+				# check if a specific error message has been received
+				if response_code in ['05', '15']:
+					raise BancardAPIPaymentMethodNotEnabledException("Bancard reported that the payer's payment method was not enabled for making payments.", bancard_response)
+				elif response_code == '12':
+					raise BancardAPIPaymentTransactionInvalidException("Bancard reported that the transaction is not valid.", bancard_response)
+				elif response_code == '51':
+					raise BancardAPIPaymentMethodNotEnoughFundsException("Bancard reported that your payment card does not have enough funds.", bancard_response)
+
+				raise BancardAPIPaymentRejectecUnknownReasonException("Bancard reported that the payment has been rejected.", bancard_response)
+
+		bancard_tx_messages = bancard_response.get('messages')[0]
+		if bancard_tx_messages.get("key", None) == 'PaymentNotFoundError':
+			# remember that you, as a marketplace, should rollback this transaction if the payer did not confirm this payment within 10 minutes
+			return False, None, bancard_response
+
+		# by default you can assume that the transaction has been rejected
+		raise BancardAPIPaymentRejectecUnknownReasonException("Bancard reported that the payment has been rejected: %s" % bancard_tx_messages.get("dsc", ""), bancard_response)
+
+	def rollback_charge(self, marketplace_charge_id, amount, currency="PYG"):
```

Comment:
The method doesn't use the amount parameter, what do you think about removing it?

Reply:
In most cases, payment processors allows you to rollback portions of the amount (not the case of Bancard). However, in the future this might be enabled and this parameter could be used for that.

Nonetheless, to simplify the code for the features provided by Bancard, I will delete it. Thanks for your comment.